### ISSUE TITLE
Bert model on a single gpu

### ISSUE THREAD

**Issue by pzajec** [thunlp/FewRel#14]
Hi!
By default settings the **pair** batch size is way to big for a single forward pass through the Bert model. How to set the parameters to train or at least evaluate such a model? (BERT + pair)

**Comment by gaotianyu1350** [comment]
You can use a smaller batch size by using the arg `--batch_size`.

**Comment by pzajec** [comment]
Sure, I can always use smaller values until it fits into memory. :)
The setting:
```
python3 train_demo.py --encoder bert --batch_size 1 --Q 1 --trainN 5 --K 3 --hidden_size 768 --model pair --pair
```
uses _14695MiB / 15079MiB_ on my gpu, while the setting:
```
python3 train_demo.py --encoder bert --batch_size 1 --Q 2 --hidden_size 768
```
uses _13691MiB / 15079MiB_.

Since it is about the trade-off between _trainN_, _N, K_ and _Q_, I was interested into a practical advice on how to set those parameters to best train the model with the current code and limited resources - single gpu. Of course at least I could do is add gradient accumulation in the training loop.

I was also interested in how did you train the models (what was the number of gpus, ...) and if you plan to release the trained BERTPair model as I would love to test in a slightly different environment.

**Comment by gaotianyu1350** [comment]
Yes, we are preparing for releasing a practical manual and our trained checkpoint really soon~

**Comment by pzajec** [comment]
Great!

**Comment by sriram-c** [comment]
Is there any trained checkpoints available for bert pair model?

**Comment by gaotianyu1350** [comment]
You can check out our updated readme (inference section)  for a BERT-PAIR checkpoint and inference instruction.
